src/relays/BasicRelay.sol

```diff
@@ -0,0 +1,97 @@
+
+pragma solidity 0.8.9;
+
+import "../WormholeGUID.sol";
+
+interface DaiJoinLike {
+    function dai() external view returns (TokenLike);
+    function exit(address, uint256) external;
+    function join(address, uint256) external;
+}
+
+interface TokenLike {
+    function balanceOf(address) external view returns (uint256);
+    function transfer(address, uint256) external returns (bool);
+}
+
+interface WormholeOracleAuthLike {
+    function requestMint(
+        WormholeGUID calldata wormholeGUID,
+        bytes calldata signatures,
+        uint256 maxFeePercentage,
+        uint256 operatorFee
+    ) external returns (uint256 postFeeAmount);
+    function wormholeJoin() external view returns (WormholeJoinLike);
+}
+
+interface WormholeJoinLike {
+    function wormholes(bytes32 hashGUID) external view returns (bool, uint248);
+}
+
+// Relay messages automatically on the target domain
+// User provides gasFee which is paid to the msg.sender
+contract BasicRelay {
+
+    DaiJoinLike            public immutable daiJoin;
+    TokenLike              public immutable dai;
+    WormholeOracleAuthLike public immutable oracleAuth;
+    WormholeJoinLike       public immutable wormholeJoin;
+
+    constructor(address _oracleAuth, address _daiJoin) {
+        oracleAuth = WormholeOracleAuthLike(_oracleAuth);
+        daiJoin = DaiJoinLike(_daiJoin);
+        dai = daiJoin.dai();
+        wormholeJoin = oracleAuth.wormholeJoin();
+    }
+
+    /**
+     * @notice Gasless relay for the Oracle fast path
+     * The final signature is ABI-encoded `hashGUID`, `maxFeePercentage`, `gasFee`, `expiry`
+     * @param wormholeGUID The wormhole GUID
+     * @param signatures The byte array of concatenated signatures ordered by increasing signer addresses.
+     * Each signature is {bytes32 r}{bytes32 s}{uint8 v}
+     * @param maxFeePercentage Max percentage of the withdrawn amount (in WAD) to be paid as fee (e.g 1% = 0.01 * WAD)
+     * @param gasFee DAI gas fee (in WAD)
+     * @param expiry Maximum time for when the query is valid
+     */
+    function relay(
+        WormholeGUID calldata wormholeGUID,
+        bytes calldata signatures,
+        uint256 maxFeePercentage,
+        uint256 gasFee,
+        uint256 expiry,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
+        require(block.timestamp <= expiry, "BasicRelay/expired");
+        bytes32 hashGUID = getGUIDHash(wormholeGUID);
+        bytes32 userHash = keccak256(abi.encode(hashGUID, maxFeePercentage, gasFee, expiry));
+        address recovered = ecrecover(userHash, v, r, s);
+        require(bytes32ToAddress(wormholeGUID.receiver) == recovered, "BasicRelay/invalid-signature");
+
+        // Initiate mint and mark the wormhole as done
+        uint256 postFeeAmount = oracleAuth.requestMint(wormholeGUID, signatures, maxFeePercentage, gasFee);
+        (,uint248 pending) = wormholeJoin.wormholes(hashGUID);
```

Comment:
I think it would be nice for `requestMint()` to return the tuple `(postFeeAmount, fee)` so that we can avoid this costly external call + sload, using `require(postFeeAmount + fee == wormholeGUID.amount)`

Reply:
Pending will already of been touched though by the `requestMint`. Does that reset to a cold SLOAD in a separate external call?

Reply:
Regardless of the hot or cold SLOAD I think it makes to return this tuple. @olivdb if you want to open a PR for that I will support.

Reply:
Good point, I did some benchmarking and the SLOAD will indeed stay hot across external calls. The external call itself adds ~1800 gas compared to getting the `fee` from the return value. Will open a PR.